aten/src/ATen/native/xnnpack/Clamp.cpp

```diff
@@ -0,0 +1,194 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/xnnpack/Allocator.h>
+#include <ATen/native/xnnpack/Common.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+namespace clamp {
+
+struct Context final {
+  Operator clamp_op;
+};
+
+namespace {
+
+// Supports NHWC and NCHW FP32
+//  - relu
+//  - relu6
+//  - leaky relu
+//  - hard tanh
+//  - or any other operator that can be expressed as a clamp
+
+bool available(
+    const int64_t input_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+         // XNNPACK
+  return xnnpack::internal::available() &&
+         // Input == Output
+         (input_channels == output_channels) &&
+         // Output Min / Max
+         (output_min.isIntegral(true) || output_min.isFloatingPoint()) &&
+         (output_max.isIntegral(true) || output_max.isFloatingPoint()) &&
+         (output_max.to<float>() > output_min.to<float>());
```

Comment:
This comparison is probably not doing what you think it is doing. if output_max is integer 2 ** 28+1 and output_min is 2**28, after being converted to float they become equal. Were they valid values to start with? It probably makes sense to pass Scalars that are already converted to float here to make it less confusing, especially since `create` already does that.

Reply:
It does guard against that, but to me it looks confusing. There will be corner cases where nominally not equal clamping parameters will be rejected because after casting to float they'll become equal, but that's unavoidable.

Reply:
OK, I'll change to float.  I don't have a strong opinion on that.

Reply:
Removing Clamp. Marking as resolved.